proposals/lip-0044.md

```diff
@@ -447,13 +419,11 @@ def getValidatorKeys(address: Address) -> ValidatorKeys:
     return validatorKeys(address)
 ```
 
-#### getGenesisData
-
-This function is used to retrieve information about the genesis data. It returns the data structure stored in the genesis data substore.
-
 #### setValidatorBLSKey
 
-This function sets the BLS key of a validator account. This update is only possible if the current BLS key of the validator is equal to a predefined constant invalid BLS key. It checks that there exists a validator account registered for the input `validatorAddress`, that the input BLS key has not been registered before in the chain, and that the input proof of possession `proofOfPossession` for the BLS key is valid. Finally, it updates the BLS key of the validator account to the input `blsKey`.
+This function sets the BLS key of a validator account. It checks that there exists a validator account registered for the input `validatorAddress`, that the input BLS key has not been registered before in the chain, and that the input proof of possession `proofOfPossession` for the BLS key is valid. Finally, it updates the BLS key of the validator account to the input `blsKey`. 
+
+**Note**: For blockchains build with the SDK v6, this function should be not be called except from the [legacy module](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0050.md#register-keys) used on the Lisk Mainchain. Otherwise, the security of interoperability may brake.
```

Comment:
```suggestion
**Note**: For blockchains build with the SDK v6, this function should be not be called except from the [Legacy module](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0050.md#register-keys) used on the Lisk Mainchain. Otherwise, the security of interoperability may brake.
```

Reply:
The last sentence it's a bit vague, maybe we could improve it a bit. I reckon the security breaks because currently we don't support updating BLS keys (maybe the reason for it it's a bit out of scope here), but what exactly would be the problem for Interoperability?

Reply:
I did not spend much time to think about it, but I would not see a security issue and think the protocol in principle should be equipped to do BLS key updates.

Reply:
To be honest, I also cannot recall the exact reason why we did not allow it from the beginning. But I thought it is (also) due to security reasons. Maybe there is indeed no issue with this. But I'm not sure we should leave it like this (without the warning) without thinking it completely through. But do you prefer to remove the statement?

Reply:
I would not remove the statement without discussing it first.